app/javascript/components/user-form/index.jsx

```diff
@@ -0,0 +1,229 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import MiqFormRenderer from '@@ddf';
+import { Loading } from 'carbon-components-react';
+import createSchema from './user-form.schema';
+import miqRedirectBack from '../../helpers/miq-redirect-back';
+import { API } from '../../http_api';
+
+const UserForm = ({ id, data }) => {
+  const [{ initialValues, isLoading }, setState] = useState({ isLoading: true });
+  const [editMode, setEditMode] = useState(false);
+  const [groups, setGroups] = useState(undefined);
+
+  useEffect(() => {
+    if (id) {
+      API.get('/api/groups?&expand=resources').then(({ resources }) => {
+        const temp = [];
+        resources.forEach((group) => {
+          temp.push({ label: group.description, value: group.id });
+        });
+        setGroups(temp);
+      }).then(() => {
+        API.get(`/api/users/${id}`).then((values) => {
```

Comment:
Hey @GilbertCherrie I know its WIP, but still, we can make use of `Promise.all` for 2 api calls here.